[kailinteoo] iP

kailinteoo opened the pull request:
### Kai

> Anyone who has never made a mistake has never tried anything new - Albery Einstein [source](https://dansilvestre.com/productivity-quotes)

**Introducing Kai: Your hassle-free task manager!**

- Text-based 
- Simple to master 
- Lightning-fast 

**Just:**
1. Download from [here](https://github.com/kailinteoo/ip/releases)
2. Double-click to launch 
3. Add your tasks 
4. Let Kassim handle the rest! 😎

**_And it's FREE!_** 

**Features:**

- [x] Add tasks, events, deadlines
- [ ] Reminders (coming soon)

***
if you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

kailinteoo committed:
IP DUKE Level-1

kailinteoo committed:
Level0

kailinteoo committed:
IP DUKE - Level 0

kailinteoo committed:
IP Level-0

kailinteoo committed:
Level-0

kailinteoo committed:
Level-1:Echo

kailinteoo committed:
Level-2: Add, List

kailinteoo committed:
Level-3:Mark as Done

kailinteoo committed:
Level-4: ToDo, Event, Deadline

kailinteoo committed:
include separator line

kailinteoo committed:
A-TextUiTesting *

kailinteoo committed:
Level-5:Handle Errors

kailinteoo committed:
Level-6: Delete

kailinteoo committed:
Level-7: Save

kailinteoo committed:
Level-7:Save

kailinteoo committed:
Level-7:save

kailinteoo committed:
Level-7:Save (edited version)

kailinteoo committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/SaveTask.java
#	src/main/java/Todo.java

kailinteoo committed:
small changes made after merging Level-7 to master branch

kailinteoo committed:
Level-8: Dates and Times

kailinteoo committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/SaveTask.java
#	src/main/java/Task.java
#	src/main/java/Todo.java

kailinteoo committed:
Level-8: Edit a bit after merching Master and Level-8 branch

kailinteoo committed:
accidentally done task 7 in this branch

kailinteoo committed:
edit of task 7 after tagging due to saving file issue

kailinteoo committed:
Refactor code to improve object-oriented design:

1. Ui Class:
   - Extracted UI-related methods from the main class into a separate Ui class.
   - The Ui class handles interactions with the user, including displaying information about marking/unmarking tasks, listing details, and removing tasks.

2. Storage Class:
   - Created a Storage class to handle loading tasks from and saving tasks to a file.
   - The class includes methods for saving tasks to a file (saveTasks), loading tasks from a file (loadTasks), and ensuring the existence of the data folder and file (ensureFolderAndFileExists).

3. Parser Class:
   - Introduced a Parser class responsible for parsing user commands.
   - The parseCommand method splits the user command and returns relevant parts, aiding in processing user inputs.

4. TaskList Class:
   - Refactored the main class to create a TaskList class, which manages the task list.
   - The TaskList class contains methods for marking/unmarking tasks, listing tasks, removing tasks, and adding new tasks based on user commands.

These changes improve code organization, readability, and maintainability by adhering to object-oriented principles and separating concerns into distinct classes. The overall structure is now more modular and extensible.

kailinteoo committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Storage.java

kailinteoo committed:
edit of storage due to conflict

kailinteoo committed:
Organize classes into 'duke' package
This commit consolidates all classes into a single package named 'duke'. This minimal package structure simplifies the organization of the codebase.

kailinteoo committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2324S2/ip

kailinteoo committed:
successfully set up gradle - can build and run Duke using gradle.

kailinteoo committed:
Applied Junit testing to a minimum of two significant methods, originating from distinct classes (Todo and Deadline). Have ensured comprehensive testing, aiming to identify and address potential issues within the targeted methods

kailinteoo committed:
Applied Junit testing to a minimum of two significant methods, originating from distinct classes (Todo and Deadline). Have ensured comprehensive testing, aiming to identify and address potential issues within the targeted methods

kailinteoo committed:
I am submitting my commits, primarily focused on creating a JAR file, updating JavaDoc, and adhering to coding standards. However, it is noteworthy that the JavaDoc and coding standard updates are predominaantly done in other branches. I have made some edits to these aspects in this branch, but the emphasis has been on the JAR file submiision.

Submitting commits with comprehensive updates, including JavaDoc enhancements and adhering to coding standards, ensures the codebase's readability, maintainability and overall quanlity.

Separating concerns into distinct branches allows for better organization and collaboration within the development team. It facilities parallel development efforts while ensuring that each aspect, such as JAR file creating receive appropriate attention and validation.

kailinteoo committed:
I have edited the JavaDoc comments within the codebase.

The JavaDoc comments require updates accurately reflect the functionality, purpose, and usage or various code elements. The ensures clarity and understanding for developers who will interact with code in future.

The focus of this commit is on revising and enhancing the JavaDoc comments to provide comprehensive documentatiion for the codebase.

Improving the JavaDoc comments helps maintain the codebase's readability and understadnability. Clear and descriptive documentation fosters easier maintenance, debugging, and collarboration among team members.

kailinteoo committed:
I have reviewed and adjusted the code to comply with the established coding standards. Adhering to coding standards promotes consistency, readability, and maintainability across the codebase. This commits focuses on making necessary adjustments to the code to align with the predefined coding standards. Maintaining consistent coding stadnards help improve code quality and reduces like the likelihood of errors and inconsistencies.

kailinteoo committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java

kailinteoo committed:
Added a new method 'find' to the 'TaskList' class to implement the search functionality
Updated the UI to display the maching tasks in the specified format
Implemented the logic to iterate through the task list and check if each task's description contains the specified  keyword

The addition of the "Find" feature enhances the usability of Duke, providing users with a convenient way to search for specific tasks within their task list. This functionality improves the overall user experience and makes it easier for users to manage their tasks effectively. This commit contributes to the ongoing development of Duke, bringing valuable new functionality to the application and enhancing its usefulness to users.

kailinteoo committed:
Merge branch 'Level-9'

kailinteoo committed:
Testing and rebugging parser class due to merging error

kailinteoo committed:
Merge branch 'A-Jar'

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

kailinteoo committed:
Debugging/Solving merging conflict between A-Jar and the master branch

danielleloh commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    /**
+     * Returns a string representation of the Deadline task.
```
Since this is an overridden method, when writing Javadocs for it, you may want to consider using `@inheritDoc` tag to reuse the header comment from the parent method but with further modifications.

kailinteoo replied:
I will take it into consideration, thankyou

danielleloh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,82 @@
+        // Initialize Scanner for user input
+        Scanner sc = new Scanner(System.in);
+        ui.blank();
+        boolean result = true;
```
Maybe you can consider renaming this boolean variable to sound like a boolean by using a name with a prefix such as `is`, `has`, `was`, etc.? Same for your other boolean method, parseCommand.

danielleloh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,201 @@
+
+    }
+
+    public void add(String command, String restOfInputs) {
```
Could this method be named with more details such that it explains what this method is doing to readers?

danielleloh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,187 @@
+     *
+     * @return The error message for an empty task description.
+     */
+    public String emptyErrorMessage() {
```
Verbs should be used for method names so maybe you could change it to sound like a verb? (for e.g. `displayEmptyErrorMessage()`). Same for your other methods.

kailinteoo replied:
okay, I have changed it

kailinteoo committed:
my Jar file isn't working, I was trying to fix it

kailinteoo committed:
Fixed my Jar file issue by rebuilding and running it

kailinteoo committed:
accidentally did my checkStyle in this branch

kailinteoo committed:
Use checkStyle to detect coding style violations

VatsalVitosh commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,76 @@
+    private void parseDeadline(String by) {
+        try {
+
+            DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+            this.by = LocalDateTime.parse(by, dateTimeFormatter);
+
+        } catch (DateTimeParseException e) {
+
+            try {
+                // If parsing as specific date/time format fails, try parsing as day of the week
+                DateTimeFormatter dayOfWeekFormatter = DateTimeFormatter.ofPattern("EEEE");
```
Consider using constants for the date formats and declaring them as private static final fields at the top of the class. Doing so would make the method more generalised and if you needed to change the format, you would only need to change the constants instead of going to each method where it's used and changing it.

Example:
```ruby
private static final String DATE_TIME_FORMAT = "d/M/yyyy HHmm";
private static final String DAY_OF_WEEK_FORMAT = "EEEE";
```

VatsalVitosh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,87 @@
+
+
+        if (command.equals("list")) {
+            TaskList taskList = new TaskList(myList);
```
Creating a new `TaskList` instance in each condition is unnecessary. Instead, perhaps create a single instance at the beginning of the method, and then use it as needed in each condition. This reduces the amount of duplicate code.

kailinteoo replied:
I have edited it accordingly. Thankyou

VatsalVitosh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,201 @@
+    public void markList(String[] parts) {
+        //System.out.println("");
+        int index = Integer.parseInt(parts[1]) - 1;
+        Ui ui = new Ui();
```
Similar to the `TaskList` instances, instead of creating a new `Ui` instance within each method, consider creating a single instance in the constructor and using it throughout the class.

kailinteoo replied:
done!

VatsalVitosh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,189 @@
+     * @param task The task that has been marked as done.
+     */
+    public void markInfo(Task task) {
+        System.out.println("------------------------------------------------------------");
```
Consider using a constant for the separation line to ensure consistency across different methods.

Example:
```ruby
private static final String SEPARATION_LINE = "------------------------------------------------------------";
```

Doing this, you won't have to copy/paste the line over and over again and can just
```ruby
System.out.println(SEPARATION_LINE);
```

VatsalVitosh reviewed (commented):
Overall the code is very well constructed and closely follows the coding standard! I've only made a few suggestions to potentially make the code a little more concise :)

kailinteoo committed:
midway creating a GUI for my chatbot

kailinteoo committed:
Created a GUi for my chatbot, but still got some minor issue regarding to the layout of the message

kailinteoo committed:
Add JavaDoc after doing the GUi and tried solving the checkstyle test issue

kailinteoo committed:
Solved the issue regarding to checkstyles and javafx error

kailinteoo committed:
Removed library code from other third party libraries

kailinteoo committed:
Removed library code (javafx-sdk-21.0.2) in my repo

kailinteoo committed:
Refactoring: Removed unnecessary commands in ui.java and duke.java

Removed unnecessary commands in ui.java and duke.java to streamline the codebase and improve readability.

Unnecessary commands clutter the code and make it harder to maintain and understand. By removing them, we ensure that the code remains concise and focused on its core functionality.

This change improves the overall code quality and makes it easier for developers to work with the UI and Duke components.

No further action is needed as the unnecessary commands have been successfully removed.

This change enhances the maintainability and readability of the codebase, making it easier for developers to contribute to and understand the project.

Avoiding unnecessary commands reduces the risk of introducing bugs and makes the codebase more robust and reliable.

kailinteoo committed:
Welcome message auto-display on SceneBuilder launch

SceneBuilder currently requires users to manually trigger the display of the welcome message upon launch. This manual step is inconvenient and breaks the flow of user interaction.

To improve user experience, implement an automatic display of the welcome message upon SceneBuilder launch.

This approach ensures that users are immediately greeted with the welcome message upon opening SceneBuilder, enhancing usability and reducing friction in the user journey.

No additional relevant information at this time.

kailinteoo committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Ui.java

kailinteoo committed:
short edit regarding to the merging of branch

kailinteoo committed:
Fix issue with creating new JAR file

Creating a new JAR file in the project currently fails due to a dependency conflict.

This approach ensures that the project can successfully generate new JAR files without encountering conflicts or errors.

No additional relevant information at this time.

kailinteoo committed:
Refactor code according to peer review feedback

Address comments provided by peers regarding the codebase:

- Update variable names for clarity and consistency.
- Improve code readability by adding comments and documentation.

This commit reflects the changes made in response to the constructive feedback received from the code review process.

kailinteoo committed:
Merge branch 'master' of https://github.com/kailinteoo/ip into A-Jar

# Conflicts:
#	data/duke.txt

kailinteoo committed:
Merge pull request #3 from kailinteoo/heads/A-Jar

Fixed my Jar file issue by rebuilding and running it

kailinteoo committed:
Introduce assertions to document important assumptions
throughout the codebase.

Assertions are added to validate critical conditions at various points in the code,
ensuring that expected conditions hold true during runtime. By using native Java assertions,
we can explicitly document these assumptions without relying solely on comments,
enhancing code clarity and maintainability.

This commit implements assertions to reinforce crucial assumptions
within the codebase.

kailinteoo committed:
Merge branch 'branch-A-Assertions'

# Conflicts:
#	src/main/java/duke/Ui.java

kailinteoo committed:
Resolve conflict in UI.java due to merging assertion branch with master

Merge of the assertion branch into the master branch resulted in
a conflict within the UI.java file.

Resolve the conflict by carefully examining the changes made in both
branches and integrating them seamlessly. Ensure that the introduction
of assertions does not disrupt the functionality or readability of
the UI.java file.

This commit addresses the conflict and ensures that the UI.java
file reflects the changes from both branches in a coherent manner.

kailinteoo committed:
Update .gitignore to exclude duke.txt file

Modify the .gitignore file to prevent the continuous committing of the
duke.txt file. This update ensures that changes to duke.txt are not
tracked by version control, thus reducing unnecessary commits related
to this file.

kailinteoo committed:
Remove unnecessary code in ui.java

Eliminate redundant or unused code from the ui.java file to improve
clarity and maintainability. This commit ensures that the codebase
remains clean and focused on essential functionality.

kailinteoo committed:
Improve code quality

Thoroughly review and refactor code to enhance quality according to
best practices. Adhere to 'Code Quality' topics for readability and
maintainability.

This commit reflects efforts to elevate overall code quality.

kailinteoo committed:
Update .gitignore to exclude duke.txt file

Modify the .gitignore file to prevent the continuous committing of the
duke.txt file. This update ensures that changes to duke.txt are not
tracked by version control, thus reducing unnecessary commits related
to this file.

kailinteoo committed:
Stop tracking duke.txt

kailinteoo committed:
Remove data/duke.txt and update gitignore

Fix the issue of duke.txt by removing it from the repository and
updating .gitignore accordingly.

This commit try to resolve the problem related to duke.txt.

kailinteoo committed:
Give help to users

Add command for accessing help page in-App.

This commit provides users with access to in-App guidance.

kailinteoo committed:
Edit ui.java for command function alignment issue

Adjust ui.java to address comments misalignment with command function.

This commit ensures comments properly align with command functions.

kailinteoo committed:
Update product name from Duke to Kai

Rename files and packages to reflect the new product name Kai.
Ensure consistency by updating UI elements to display the correct product name.

This commit implements the change from "Duke" to "Kai" throughout
the project to maintain coherence.

kailinteoo committed:
Set up product website

Add Ui.png to docs folder with full GUI window screenshot. Update
README.md with brief User Guide.

This commit completes the setup of the product website.

kailinteoo committed:
Update README with product information and usage instructions

This commit enhances the README with details about the product
and provides concise instructions for users.

kailinteoo committed:
Readjust Ui.png for better representation

Updated Ui.png to ensure accurate representation of the product's GUI.

kailinteoo committed:
Change jar filename in build.gradle

Update jar filename in build.gradle to reflect new naming convention.

kailinteoo committed:
Update README and UI.java

Fix errors in README message and edit UI.java file.

kailinteoo committed:
Refactor code to improve quality

Remove unnecessary code to enhance readability and maintainability.
This commit focuses on improving code quality by eliminating redundant
or obsolete code segments.

kailinteoo committed:
Update README with product information and usage instructions

This commit enhances the README with details about the product
and provides additional instructions for users.
